### DISCUSSION TITLE
How to load subtitles? Solution and example required.

### DISCUSSION THREAD

**Discussion by TW2** [gstreamer-java/gst1-java-core#246] [General]
I try to load an external subtitles file like that:
```
public void setAssFile(File assFile) {
        playbin.set("suburi", assFile.toURI());
        
        Set<PlayFlags> flags = playbin.getFlags();
        flags.clear();
        
        flags.add(PlayFlags.VIDEO);
        flags.add(PlayFlags.AUDIO);
        flags.add(PlayFlags.TEXT);
        
        playbin.setFlags(flags);
}
```

The 'suburi' returns the good uri for subtitles and when playing the content is transfered to 'current-suburi' perfectly.
At playing 'n-text' is '1' and 'current-text' is '0' (my video source has no subtitles).

But there is no subtitles in video display! How to do it?

Note: I precise that a video source muxed with subtitles displays its default subtitles object during playback.

**Comment by neilcsmith-net** [comment]
Moving this over to discussions. We normally ask for support queries to go on the mailing list, but am considering using Discussions here too.

What sink are you using on your PlayBin?

**Comment by TW2** [comment]
I don't create a sink by myself. Maybe the default is loaded.

**Comment by neilcsmith-net** [comment]
In which case, subtitles should usually display OK.  Does it work from the command line?

**Comment by TW2** [comment]
For this case : 
```
C:\Dev\gstreamer\1.0\mingw_x86_64\bin\gst-launch-1.0 -v textoverlay name=ov ! autovideosink filesrc location=media.mkv ! decodebin ! ov.video_sink filesrc location=sub.ass ! subtitleoverlay ! ov.text_sink
```
I can see a video without audio and without subtitles. The subtitles has been created with Aegisub.

May I forgotten something ?

[DESKTOP-RA92GMO.txt](https://github.com/gstreamer-java/gst1-java-core/files/7687660/DESKTOP-RA92GMO.txt)

Note : I use Windows 11 Pro Insiders

**Reply by neilcsmith-net** [reply]
But that's not using PlayBin. Try using `gst-launch-1.0` with just a playbin element and the properties you've set in Java code. Does it work or not?  If not, I suggest moving the discussion to GStreamer's own mailing lists.  If you do get a working playbin there, then we need to look at what is different.

**Reply by TW2** [reply]
My ```C:\Dev\gstreamer\1.0\mingw_x86_64\bin\gst-launch-1.0 playbin uri=file:///C:/Users/util2/Desktop/test_gst/media.mkv``` works with video and audio and I don't know how to add subs. But in fact playbin is okay.

**Reply by neilcsmith-net** [reply]
Same way you tried in Java - `suburi=file://....`  Also see `gst-inspect-1.0 playbin`.  If that doesn't work, then time to ask at https://lists.freedesktop.org/mailman/listinfo/gstreamer-devel

**Reply by TW2** [reply]
```gst-launch-1.0 playbin uri=file:///C:/Users/util2/Desktop/test_gst/media.mkv suburi=file:///C:/Users/util2/Desktop/test_gst/sub.ass``` plays video and audio only

```gst-launch-1.0 playbin uri=file:///C:/Users/util2/Desktop/test_gst/mediaWithSubs.mkv``` (with subs embedded) is okay.

**Reply by neilcsmith-net** [reply]
In which case, I suggest asking on the upstream mailing list. It suggests GStreamer doesn't like the subtitles for some reason. We can only help with the Java bindings.

**Reply by TW2** [reply]
Ok, thanks a lot.

**Reply by TW2** [reply]
Hello, I had tested the way to add subtitles.
SRT ok, MKV with ASS ok.

No way to open an ASS from file because it's not implemented like that.
ASS is only working on MKV with ASS subtitles that are already inside and one track (we can have multiple tracks here) is loaded automatically on playing. Thanks for your help ! My apologies. We had to use an external ASS parser to put it on video image or on overlay.
